### PR TITLE
added login authentication logic

### PR DESCRIPTION
**Pull request by Tanvir04khan**

### CODE REVIEW

File: .firebase/hosting.YnVpbGQ.cache [hunk 1]
```diff
@@ -0,0 +1,17 @@
+favicon.ico,499162500000,eae62e993eb980ec8a25058c39d5a51feab118bd2100c4deebb2a9c158ec11f9
```
Comment (faran4engg) [inline .firebase/hosting.YnVpbGQ.cache:1]: add it in gitignore `*.cache`

File: src/components/Login/LoginScreen.jsx [hunk 2]
```diff
@@ -1,16 +1,37 @@
...
   return (
     <div className="login">
       <div className="login__container">
+        <h1>Youtube</h1>
         <img
           src="https://www.freeiconspng.com/thumbs/youtube-logo-png/hd-youtube-logo-png-transparent-background-20.png"
           alt="Youtube logo"
         />
-        <button>Login with google</button>
+        <button onClick={loginHandler}>Login with google</button>
```
Comment (faran4engg) [inline src/components/Login/LoginScreen.jsx:34]: follow one convention
either loginHandler or handleLogin

File: src/firebase.jsx [hunk 3]
```diff
@@ -1,12 +1,16 @@
 import firebase from "firebase/app";
```
Comment (faran4engg) [inline src/firebase.jsx]: why .jsx ?

File: src/redux/action/auth.action.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+import firebase from "firebase/app";
+import auth from "../../firebase";
+import {
+  LOGIN_REQUEST,
+  LOAD_PROFILE,
+  LOGIN_SUCCESS,
+  LOGIN_FAIL,
+  LOG_OUT,
+} from "../actionType";
```
Comment (faran4engg) [inline src/redux/action/auth.action.jsx]: or you can do, import * as ActionTypes. then use it as ActionTypes.abc

File: src/redux/action/auth.action.jsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+export const login = () => async (dispatch) => {
+  try {
+    dispatch({
+      type: LOGIN_REQUEST,
+    });
+
+    const provider = new firebase.auth.GoogleAuthProvider();
+    const response = await auth.signInWithPopup(provider);
+    console.log(response);
```
Comment (faran4engg) [inline src/redux/action/auth.action.jsx]: remove this

File: src/redux/reducer/auth.reducer.jsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  LOGIN_REQUEST,
+  LOAD_PROFILE,
+  LOGIN_SUCCESS,
+  LOGIN_FAIL,
+} from "../actionType";
+
+const initialState = {
+  accessToken: sessionStorage.getItem("access-token")
+    ? sessionStorage.getItem("access-token")
+    : null,
```
Comment (faran4engg) [inline src/redux/reducer/auth.reducer.jsx]: i dont think u need ternary here

**Review by faran4engg — APPROVED** [review]
minor comments

File: src/components/Login/LoginScreen.jsx (hunk 2 above)
Comment (Tanvir04khan) [reply src/components/Login/LoginScreen.jsx:34]: changed code as per suggestion

File: src/firebase.jsx (hunk 3 above)
Comment (Tanvir04khan) [reply src/firebase.jsx]: changed file name as per suggestion

File: src/redux/action/auth.action.jsx (hunk 4 above)
Comment (Tanvir04khan) [reply src/redux/action/auth.action.jsx]: changed code as per suggestion

File: src/redux/action/auth.action.jsx (hunk 5 above)
Comment (Tanvir04khan) [reply src/redux/action/auth.action.jsx]: changed code as per suggestion

File: src/redux/reducer/auth.reducer.jsx (hunk 6 above)
Comment (Tanvir04khan) [reply src/redux/reducer/auth.reducer.jsx]: changed code as per suggestion

**Pull request merged**